core/src/main/java/org/carbondata/scan/collector/impl/AbstractScannedResultCollector.java

```diff
@@ -170,33 +120,25 @@ private Object getMeasureData(MeasureColumnDataChunk dataChunk, int index, DataT
   /**
    * Below method will used to get the result
    */
-  @Override public Result getCollectedResult() {
-    Result<List<ListBasedResultWrapper>, Object> result = new ListBasedResult();
-    if (tableBlockExecutionInfos.isFixedKeyUpdateRequired() && tableBlockExecutionInfos
-        .isDimensionsExistInQuery()) {
-      updateKeyWithLatestBlockKeygenerator();
-      result.addScannedResult(listBasedResult);
-    } else {
-      result.addScannedResult(listBasedResult);
+  protected void updateData(List<Object[]> listBasedResult) {
+    if (tableBlockExecutionInfos.isFixedKeyUpdateRequired()) {
+      updateKeyWithLatestBlockKeygenerator(listBasedResult);
     }
-    return result;
   }
 
-
-
   /**
    * Below method will be used to update the fixed length key with the
    * latest block key generator
    *
    * @return updated block
    */
-  private void updateKeyWithLatestBlockKeygenerator() {
+  private void updateKeyWithLatestBlockKeygenerator(List<Object[]> listBasedResult) {
```

Comment:
Please correct me if i m wrong but now we are from executor layer for dictionary columns we are sending surrogate keys ..key updation is required only when aggregation is done is executor ..this is because of dynamic cardinality. As now from DimensionColumnDataChunk we are filling the surrogate key ,so now do we need this method ??